### PR TITLE
Use netrunner-cards-json container

### PR DESCRIPTION
**Pull request by leukosaima**
Workflow to push to DO registry, docker-compose that uses the netrunner-cards-json container

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,17 +17,33 @@ RUN mkdir -p $RAILS_ROOT/tmp/pids
...
-RUN pwd
+COPY Gemfile Gemfile.lock $RAILS_ROOT/
 
-COPY Gemfile Gemfile.lock ./
+# Install gems into the vendor/bundle directory in the workspace.
+RUN bundle config set --local path "vendor/bundle" && \
+  bundle install --jobs 4 --retry 3
 
-RUN bundle install
+# Precompile app assets as the final step.
```
Comment (leukosaima) [inline Dockerfile:29]: no precompiling happening, bad comment

File: docker-compose.override.dev.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  nrdb_api_server:
+    image: registry.digitalocean.com/nsgcr/nrdb_api_server:dev
+    # build the Dockerfile next to this file, tag the built image with above image name
+    build: .
+    command:
+      - /bin/sh
+      - -c
+      - |
+        rake db:reset
+        bundle exec rails cards:import
```
Comment (leukosaima) [inline docker-compose.override.dev.yml:18]: To be discussed, this resets the db and re-imports cards everytime, which could be fine for dev workflows, it just takes a few more seconds to wait for everything to be ready after a start, or restart of the containers

File: Dockerfile [hunk 3]
```diff
@@ -13,17 +17,33 @@ RUN mkdir -p $RAILS_ROOT/tmp/pids
...
+COPY . $RAILS_ROOT/
```
Comment (leukosaima) [inline Dockerfile:30]: Probably should only copy what is needed

File: docker-compose.yml [hunk 4]
```diff
@@ -20,13 +30,14 @@ services:
...
 
-    # makes the app container aware of the DB container
-    links:
+    # starts the app container after the DB and netrunner_cards_json containers
+    depends_on:
       - db
+      - netrunner_cards_json
 
     volumes:
-      - .:/var/www/nrdb-api/
```
Comment (leukosaima) [inline docker-compose.yml:28]: @plural I suppose you were using this to "hot reload" some of the files while the container is running ?

Comment (plural) [reply docker-compose.yml:28]: exactly, for dev purposes.  i can use a devcontainer for that instead.

**Review by plural — COMMENTED** [review]
please set your editor to add newlines to your files on save.  Several of this are missing them.

i'm going to try this out in a bit

**Pull request merged**